backend/api/professor/grading_materials.py

```
@@ -0,0 +1,333 @@
+from middleware.role_auth import require_professor
+from services.grading_embedding_service import grading_embedding_service
+from utilities.debug_logging import debug_log
+
+router = APIRouter(prefix="/professor", tags=["Grading Materials"])
+
+@router.post("/simulations/{simulation_id}/grading-materials")
+async def upload_grading_materials(
+    simulation_id: int,
+    background_tasks: BackgroundTasks,
+    file: UploadFile = File(...),
+    description: Optional[str] = Form(None),
```

Ruff (0.14.1)

Unused function argument: `description`

(ARG001)